### ISSUE TITLE
Assigning a HasMany ids field to one item crashes

### ISSUE THREAD

**Issue by CyborgMaster** [composite-primary-keys/composite_primary_keys#290]
If I have a model with a `has_many` association and I assign its related `ids=` method to a single item, it crashes.  If it assign it to multiple items it works just fine.  This problem does not occur for associations that don't use CPK.  

I have models that look like this:
``` ruby
class Species
  self.primary_keys = [:species_id, :source_id]
  has_many :patients, foreign_key: [:species_id, :source_id]
end

class Patient
  self.primary_keys = [:patient_id, :source_id]
  belongs_to :species, foreign_key: [:species_id, :source_id]
end
```

When I run the following in the rails console, it works fine (the id's match up to test data in my dev DB):
``` ruby
Species.first.patient_ids = [['10000',1],['10001',1]]
```

However, this crashes:
``` ruby
Species.first.patient_ids = [['10000',1]]
```
Here's the backtrace:
```
NoMethodError: undefined method `index_by' for #<Patient:0x0000010eb77578>
0: /Users/jeremy/.rvm/gems/ruby-2.0.0-p451@allynova/gems/activemodel-4.2.0/lib/active_model/attribute_methods.rb:433:in `method_missing'
1: /Users/jeremy/.rvm/gems/ruby-2.0.0-p451@allynova/bundler/gems/papercrop-572f215f70bf/lib/papercrop/model_extension.rb:82:in `method_missing'
2: /Users/jeremy/.rvm/gems/ruby-2.0.0-p451@allynova/gems/activerecord-4.2.0/lib/active_record/associations/collection_association.rb:67:in `ids_writer'
3: /Users/jeremy/.rvm/gems/ruby-2.0.0-p451@allynova/gems/activerecord-4.2.0/lib/active_record/associations/builder/collection_association.rb:76:in `patient_ids='
4: (pry):2:in `<main>'
```

**Comment by cfis** [comment]
Thanks for the report.  This error is due to the way CPK ids are parsed.  Hopefully fixed d37cb84.

**Comment by CyborgMaster** [comment]
Glad to be of help.  I'll try and test the fix here sometime soon and verify on my end that it resolves the problem.  

Thanks for the fix.
